### ISSUE TITLE
NoCredentialProviders: no valid providers in chain

### ISSUE THREAD

**Issue by echarles** [kubernetes-sigs/cluster-api-provider-aws#358]
/kind bug

**What steps did you take and what happened:**

Step 1: build from source

```bash
cd $GOPATH/src/sigs.k8s.io/cluster-api-provider-aws && \
  make clean && \
  make dep-ensure && \
  make cluster-api-dev-helper && \
  make clusterawsadm && \
  make clusterctl && \
  cp bazel-bin/cmd/clusterawsadm/linux_amd64_pure_stripped/clusterawsadm /usr/local/bin/clusterawsadm && \
  cp bazel-bin/cmd/clusterctl/linux_amd64_pure_stripped/clusterctl /usr/local/bin/clusterctl
```

Step 2 - Create the AWS prerequisites (wiith a local user defined in ~/.aws)

```bash
cd $GOPATH/src/sigs.k8s.io/cluster-api-provider-aws && \
  export AWS_REGION=us-east-1 && \
  aws ssm put-parameter --name "/sigs.k8s.io/cluster-api-provider-aws/ssh-key" \
    --type SecureString \
    --value "$(aws ec2 create-key-pair --key-name cluster-api-provider-aws.sigs.k8s.io | jq .KeyMaterial -r)" && \
  clusterawsadm alpha bootstrap create-stack && \
  export AWS_CREDENTIALS=$(aws iam create-access-key --user-name bootstrapper.cluster-api-provider-aws.sigs.k8s.io) && \
  export AWS_ACCESS_KEY_ID=$(echo $AWS_CREDENTIALS | jq .AccessKey.AccessKeyId -r) && \
  export AWS_SECRET_ACCESS_KEY=$(echo $AWS_CREDENTIALS | jq .AccessKey.SecretAccessKey -r) && \
  export SSH_KEY_NAME="cluster-api-provider-aws.sigs.k8s.io" && \
  cd $GOPATH/src/sigs.k8s.io/cluster-api-provider-aws && \
  make manifests && \
  ls cmd/clusterctl/examples/aws/out
```

Step 3 - Create the Cluster

```bash
  clusterctl create cluster \
    -v6 \
    --logtostderr \
    --vm-driver kvm2 \
    --provider aws \
    -m ./cmd/clusterctl/examples/aws/out/machines.yaml \
    -c ./cmd/clusterctl/examples/aws/out/cluster.yaml \
    -p ./cmd/clusterctl/examples/aws/out/provider-components.yaml
```

Step 4 - Collect logs

```bash
kubectl get po -o name -n aws-provider-system | grep aws-provider-controller-manager | xargs kubectl logs -n aws-provider-system -c manager -f
```

shows:

```
go:64] Reconciling cluster test1.
W1103 08:23:54.597464       1 controller.go:175] unable to create machine aws-controlplane-nw4x6: failed to create or get machine: failed to query instance by tags: failed to describe instances by tags: NoCredentialProviders: no valid providers in chain. Deprecated.
        For verbose messaging see aws.Config.CredentialsChainVerboseErrors
{"level":"error","ts":1541233434.597651,"logger":"kubebuilder.controller","caller":"controller/controller.go:209","msg":"Reconciler error","Controller":"machine-controller","Request":"default/aws-controlplane-nw4x6","error":"failed to create or get machine: failed to query instance by tags: failed to describe instances by tags: NoCredentialProviders: no valid providers in chain. Deprecated.\n\tFor verbose messaging see aws.Config.CredentialsChainVerboseErrors","errorVerbose":"NoCredentialProviders: no valid providers in chain. Deprecated.\n\tFor verbose messaging see aws.Config.CredentialsChainVerboseErrors\nfailed to describe instances by tags\nsigs.k8s.io/cluster-api-provider-aws/p
...

**Environment:**

- Cluster-api-provider-aws version: build from e74f4f9
- Kubernetes version: (use `kubectl version`):  
- OS (e.g. from `/etc/os-release`): Ubuntu 18.04.1
- Minikube v0.30.0

**Comment by detiber** [comment]
@echarles it looks like from the #cluster-api-aws backlog that you were able to get through this issue, could you provide a followup with how you were able to resolve the issue so that we can work on improving the documentation?

**Comment by echarles** [comment]
@detiber sure, I was going to open a PR for doc enhancement. My experience so far is that enough permission must be provided to the user defined in the `cmd/clusterctl/examples/aws/out/credential` file.

Quick feedback:

1. I anticipate that the permissions provisioned via the `clusterawsadmin` command are not enough because, when I am using a self-created user with "quite large" permissions, the cluster is created.

2. I have also encountered #241 but also think it was due to insufficient permissions:

3. The particular issue opened here `NoCredentialProviders` was due to an empty  `cmd/clusterctl/examples/aws/out/credential` (I probably created the manifests from the bad terminal without env var) - This can thus be closed.

To follow-up I will open that doc PR tomorrow (will also cover your comment on #359).
